### DISCUSSION TITLE
safer version of hidden field

### DISCUSSION THREAD

**Discussion by jdion84** [filamentphp/filament#9897] [Help]
right now i have this hidden field:

```php
                Hidden::make('type')
                    ->default(UserType::Admin),
```

however, docs say user can edit this via browser dev tools

what would be a safer approach? something like this?:

```php
                Hidden::make('type')
                    ->dehydrateStateUsing(fn () => UserType::Admin),
```

**Comment by devhoussam1998** [comment]
the also said _**You should not use this component to store sensitive or read-only information**_ can you explain more about

`Hidden::make('type')` what is your expected result?

**Reply by jdion84** [reply]
i think my `dehydrateStateUsing` example might be fine because it will always set the value to what i want even if the user edits the hidden field client side
